pyemu/pst/pst_handler.py

```diff
@@ -3005,3 +2804,29 @@ def get_adj_pars_at_bounds(self, frac_tol=0.01):
 
         return under_lb,over_ub
 
+    def try_parse_name_metadata(self):
+        """try to add meta data columns to parameter and observation data based on
+        item names.  Used with the PstFrom process.
+
+        Note: metadata is identified in key-value pairs that are separated by a colon.
+            each key-value pair is separated from others by underscore
+
+        """
+        par = self.parameter_data
+        obs = self.observation_data
+        par_cols = pst_utils.pst_config["par_fieldnames"]
+        obs_cols = pst_utils.pst_config["obs_fieldnames"]
+
+        for df,name,fieldnames in zip([par,obs],["parnme","obsnme"],[par_cols,obs_cols]):
+            items = df.loc[:,name].apply(lambda x: [item.split(':') for item in x.split('_') if ':' in item])
+            meta_dict = items.apply(lambda x: {k:v for xx in x for k,v in x})
```

Comment:
Some of these apply functions can get suuuuuper slow when the DataFrames are long. I think that these can be condensed into one apply.
```suggestion
            meta_dict = df.loc[:,name].apply(lambda x: dict([item.split(':') for item in x.split('_') if ':' in item]))
```
?

Reply:
nice!  I had them combined originally, but, in the course of debugging my hackery, I undid it - sad

Reply:
nailed it